test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ScalaJSDefinedTest.scala

```diff
@@ -888,6 +964,66 @@ object ScalaJSDefinedTest extends JasmineTest {
       expect(dyn.dependent(8)).toEqual(-1)
     }
 
+    it("constructors with default parameters (ScalaJSDefined/-)") {
+      expect(new ConstructorDefaultParam0().foo).toEqual(0)
```

Comment:
What about giving meaningful names to those, e.g., `ConstructorDefaultParam_JSNonNative_None` with choices being `Scala`, `JSNonNative`, `JSNative` and `None`?